### ISSUE TITLE
Dict ordering isn't preserved 

### ISSUE THREAD

**Issue by mustafa0x** [sdispater/tomlkit#180]
```py
# Python 3.9.7

In [1]: import tomlkit as toml

In [2]: d = {
   ...:     "id": "<id>",
   ...:     "meta": {
   ...:         "name": "name",
   ...:         "desc": "",
   ...:         "created": "now",
   ...:     },
   ...:     "cards": [1, 2, 3],
   ...: }

In [3]: toml.dumps(d)
Out[3]: 'id = "<id>"\ncards = [1, 2, 3]\n\n[meta]\nname = "name"\ndesc = ""\ncreated = "now"\n'

In [5]: import collections

In [6]: d1 = collections.OrderedDict(d)

In [7]: toml.dumps(d1)
Out[7]: 'id = "<id>"\ncards = [1, 2, 3]\n\n[meta]\nname = "name"\ndesc = ""\ncreated = "now"\n'

In [8]: toml.__version__
Out[8]: '0.10.0'

In [9]: d1 = collections.OrderedDict(d)

In [10]: d1
Out[10]: 
OrderedDict([('id', '<id>'),
             ('meta', {'name': 'name', 'desc': '', 'created': 'now'}),
             ('cards', [1, 2, 3])])

In [11]: toml.dumps(d1)
Out[11]: 'id = "<id>"\ncards = [1, 2, 3]\n\n[meta]\nname = "name"\ndesc = ""\ncreated = "now"\n'

In [12]: import json

In [13]: json.dumps(d)
Out[13]: '{"id": "<id>", "meta": {"name": "name", "desc": "", "created": "now"}, "cards": [1, 2, 3]}'
```

**Comment by frostming** [comment]
How should the order be preserved? By default, dicts are converted to toml tables and top level key-value pairs can't appear after a table header, otherwise it will be recognized as the key belonging to the table. So the following render result doesn't equal to the original data:

```toml
id = "<id>"

[meta]
name = "name"
desc = ""
created = "now"

card = [1, 2, 3]  # card belongs to [meta] table
```

If you want to render `meta` table as an inline table instead, you must do it yourself rather than feeding with a plain dict object.

**Comment by mustafa0x** [comment]
Thanks.

**Comment by frostming** [comment]
It was due to a sort bug fixed by 5a61e5b5527a1649fdad46b17dee585d2237d041

Still, you can make it work by constructing the document manually(`tomlkit.table()` -> `tomlkit.document()`):

```python
meta = toml.inline_table()
meta.update(
    {
        "name": "name",
        "desc": "",
        "created": "now",
    }
)

d = toml.document()
d.add("id", "<id>")
d.add("meta", meta)
d.add("cards", [1, 2, 3])

print(toml.dumps(d))
```

gives

```toml
id = "<id>"
meta = {name = "name", desc = "", created = "now"}
cards = [1, 2, 3]
```
